src/rubicon/objc/api.py

```diff
@@ -1627,17 +1571,22 @@ def _load_methods(self):
 
         methods_ptr = libobjc.class_copyMethodList(self, byref(methods_ptr_count))
 
-        if self.superclass is not None:
-            if self.superclass.methods_ptr is None:
-                with self.superclass.cache_lock:
-                    self.superclass._load_methods()
+        # Traverse superclasses and load methods.
```

Comment:
I am not entirely sure while a full traversal of superclasses is required now but wasn't previously for tests on Python 3.12 + macOS 15 to pass. But I do believe that a class hierarchy travels makes sense regardless.

Reply:
Agreed this is weird; I can only assume we're hitting some weird cache initialisation order thing (e.g., the test was previously initializing the super class before the class that was causing a problem). However, the solution here makes sense.

Reply:
Running the toga testbed suite reveals [a similar issue](https://github.com/beeware/toga/actions/runs/12088971183/job/33713572376?pr=2978) for `TogaSlider.setValue_animated_` where the superclass method cannot be found. This is despite the fix here and could be related to the TogaSlider being a subclass defined in Python.

Reply:
Note that the direct method call `TogaSlider.setValue_animated(value, True)` succeeds.

Reply:
I've managed to fix that by allowing direct method lookup if there is no cached partial method, similar to what we already do for the old style syntax.

This is very likely a race condition somewhere, but the code around `ObjCClass._load_methods`, `ObjCClass._cache_method` and `ObjCPartialMethod` quite complex and hard to follow.

Reply:
But all those changes to method lookup are making this PR a bit unwieldy. I've reverted some of the unneeded commits and am happy to split this off into an entirely different PR if that makes it easier.

Reply:
Well, something is still very fishy about my solution in this PR. `_load_methods` should have already been recursive, with the main difference that is was hard-stopping the recursion if one of the superclasses had already called `_load_methods`. This makes assumptions of the superclass chain not changing during `init`.

The new solution forces recursion to continue, bad in a horribly hacky way.

Reply:
Ok, I have had another look at all those recursive calls and I think that (1) I do understand better now what is going on and (2) have a more elegant solution than this PR to method loading fixes (https://github.com/samschott/rubicon-objc/tree/method-loading).

But this is indeed better reviewed in isolation, so I am happy to proceed with this PR as is and send out an independent PR for stylistic improvements and hopefully clearer code logic.